### PR TITLE
Fix broken unit tests

### PR DESCRIPTION
**Pull request by aspandey**
Added fake deployment for mcg consol to fix the failure of tests.

### CODE REVIEW

**Comment by aspandey** [comment]
@rexagod  @naveenpaul1  Please review the code and provide your comments on this.

File: controllers/managedmcg_controller_test.go [hunk 1]
```diff
@@ -72,6 +83,17 @@ func newManagedMCGFake() mcgv1alpha1.ManagedMCG {
 	return managedMCGFake
 }
 
+func newConsoleDeploymentFake() appsv1.Deployment {
+	ConsoleDepFake := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mcg-ms-console",
+			Namespace: "openshift-storage",
```
Comment (rexagod) [inline controllers/managedmcg_controller_test.go:90]: `openshift-storage`? Do we want this to fail? If we are trying to cover Vinayak's PR, I'd suggest doing so for the happy path first, then the corner cases. LMK WDYT.

**Review by rexagod — COMMENTED** [review]
There's two things I'm confused about.

- This PR targets "broken" unit tests. But, if you look at the CI, we don't have any broken **unit** tests at the moment: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/red-hat-storage_mcg-osd-deployer/44/pull-ci-red-hat-storage-mcg-osd-deployer-main-test-suite/1532286598601773056#1:build-log.txt%3A122-126, am I missing something?
- Assuming this PR aims to fix the bug that's in master right now, it should target the integration test that's (still) failing, as seen here (CI logs for this PR): https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/red-hat-storage_mcg-osd-deployer/44/pull-ci-red-hat-storage-mcg-osd-deployer-main-test-suite/1532286598601773056#1:build-log.txt%3A146

Again, I could be way off on what's intended here, PMLK.

**Comment by aspandey** [comment]
> There's two things I'm confused about.
> 
> * This PR targets "broken" unit tests. But, if you look at the CI, we don't have any broken **unit** tests at the moment: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/red-hat-storage_mcg-osd-deployer/44/pull-ci-red-hat-storage-mcg-osd-deployer-main-test-suite/1532286598601773056#1:build-log.txt%3A122-126, am I missing something?

Unit tests started failing after merging few patches in past. One of those were the patch sent by Vinayak for console deployment.
I have fixed it in Unit test and then I realized that E2E test are also failing for the same reason. I need to resend this patch including fix for E2E tests also.
You are not seeing "broken test" in this patch because this patch is fixing it.
You can just clone the current master branch and try to run unit and E2E tests and they will fail.
Ideally this should have been captured when the patch was sent which caused this failure but it did not happen.


> * **Assuming** this PR aims to fix the bug that's in master right now, it should target the integration test that's (still) failing, as seen here (CI logs for this PR): https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/red-hat-storage_mcg-osd-deployer/44/pull-ci-red-hat-storage-mcg-osd-deployer-main-test-suite/1532286598601773056#1:build-log.txt%3A146
> 
> Again, I could be way off on what's intended here, PMLK.

File: controllers/managedmcg_controller_test.go (hunk 1 above)
Comment (aspandey) [reply controllers/managedmcg_controller_test.go:90]: Why are you thinking it should not be "openshift-storage"? I am targeting happy path only...May be I am missing something.

By the way, I am not very good in short forms :) :)  "LMK WDYT"?? I spent more time in finding its meaning :)

**Comment by rexagod** [comment]
> You are not seeing "broken test" in this patch because this patch is fixing it.

Ah okay, my bad. Are you going to send the e2e fix to this PR as well? Or a different one?

File: controllers/managedmcg_controller_test.go (hunk 1 above)
Comment (rexagod) [reply controllers/managedmcg_controller_test.go:90]: I'm assuming deployment should ideally be under `redhat-data-federation`? Is that not the case?

**Comment by aspandey** [comment]
> > You are not seeing "broken test" in this patch because this patch is fixing it.
> 
> Ah okay, my bad. Are you going to send the e2e fix to this PR as well? Or a different one?

I am thinking of with this patch only...But if it takes time then I will send different patch. I will try to do it by EOD.

File: controllers/managedmcg_controller_test.go (hunk 1 above)
Comment (aspandey) [reply controllers/managedmcg_controller_test.go:90]: "redhat-data-federation" we use in E2E testing.
I understand the confusion :)

**Comment by rexagod** [comment]
Awesome, that would be nice. Thanks!

**Comment by aspandey** [comment]
@naveenpaul1 I have committed my "Work in Progress" code..you can take a look ...

**Comment by aspandey** [comment]
These changes has been merged as part of different patch sent by Pranshu.
Closing it now.
